GitCommands/CommitDataManager.cs

```diff
@@ -199,8 +203,13 @@ private bool TryGetCommitLog(string commitId, string format, [NotNullWhen(return
                 commitId
             };
 
-            // Do not cache this command, since notes can be added
-            data = GetModule().GitExecutable.GetOutput(arguments, outputEncoding: GitModule.LosslessEncoding);
+            // This command can be cached if commitId is a git sha and Notes are ignored
+#if DEBUG
+            Debug.Assert(!cache || ObjectId.TryParse(commitId, out _), $"Cache should be used only for sha");
+#endif
```

Comment:
`Debug.Assert` will only work in debug
```suggestion
            Debug.Assert(!cache || ObjectId.TryParse(commitId, out _), $"Cache should be used only for sha");
```

Reply:
I originally had more asserts, but even this is maybe more CPU intensive then required
It may be removed later.